libr/bin/format/elf/elf.c

```diff
@@ -2718,28 +2719,32 @@ static size_t populate_relocs_record_from_section(ELFOBJ *bin, RBinElfReloc *rel
 		size = get_size_rel_mode (rel_mode);
 
 		for (j = get_next_not_analysed_offset (bin, bin->g_sections[i].offset, 0, bin->baddr);
-			j < bin->g_sections[i].size;
+			j < bin->g_sections[i].size && p < num_relocs;
 			j = get_next_not_analysed_offset (bin, bin->g_sections[i].offset, j + size, bin->baddr)) {
-
-			if (!read_reloc (bin, relocs + pos, rel_mode, bin->g_sections[i].offset + j)) {
+			if (!j) {
```

Comment:
why this? I don't see why `get_next_not_analysed_offset` should return 0 and why that should be a "invalid" value.

Reply:
yeah that was a weak hacky chk to avoid trashy relocs when parsing crap. it "works" for my case, but it was a clearly wrong check. any proposal on how to properly check for the loop end?

Reply:
i have no idea what trashy relocs you are talking about, so no, i have no idea, but this looks wrong. Please provide test case so we can understand what you are referring to.